### DISCUSSION TITLE
problem in fx.init('default', col_names = collaborator_list) ---

### DISCUSSION THREAD

**Discussion by amin-kaveh** [securefederatedai/openfederatedlearning#784] [Q&A]
Hi,

When I run the available tutorials and examples in openFl (for pyTorch), it perfectly works. For example the following code is working fine:
```
... 
import openfl
import openfl.native as fx
from openfl.federated import federatedModel, FederatedDataSet
...
# read train and valid data with torchvision(MNIST)
# define Net, optimizer and loss function
...
collaborator_list = ['one', 'two']
fx.init('torch_cnn_mnist', col_names = collaborator_list)

fl_data = FederatedDataSet(train_img, train_labels, valid_img,valid_labels, batch_size =32, num_classes = 10)
fl_model = FederatedModel(build_model = Net, optimizer = optimizer, loss_fn = cross_entropy, data_loader = fl_data)

experiment_collaborators = {col_name:col_model for col_name, col_model in zip(collaborator_list, fl_model.setup(2))}
final_fl_model = fx.run_experiment(experiment_collaborators, override_config ={"aggregator.settings.round_to_train": 5})
```

and this code perfectly works.

Now suppose I just change `fx.init('torch_cnn_mnist', col_names = collaborator_list)` to `fx.init('default', col_names = collaborator_list)`, and the rest of the code is the same as before. But I get the following error:

```
Traceback (most recent call last):
  File "/home/amin/myCode.py", line 150, in <module>
    final_fl_model = fx.run_experiment(experiment_collaborators, override_config={"aggregator.settings.rounds_to_train": 5})
  File "/home/amin/myCode.py", line 295, in run_experiment
    collaborator.run_simulation()
  File "/home/amin/lib/python3.9/site-packages/openfl/component/collaborator/collaborator.py", line 170, in run_simulation
    self.do_task(task, round_number)
  File "/home/amin/lib/python3.9/site-packages/openfl/component/collaborator/collaborator.py", line 255, in do_task
    global_output_tensor_dict, local_output_tensor_dict = func(
TypeError: train() got an unexpected keyword argument 'col_name'

```

Why is this happening and how I can fix it. I am asking because I want to perform a federated learning on a different dataset and a different network model (FeedForward NN) and I need to know how in general I can run federated learning with openfl.

**Comment by daviddelriod** [comment]
I suggest you try using the **python native api** instead of trying to reshape one of the tutorials to you custom project. I have just finished my FL experiment and at first I also tried to replicate the tutorials' examples but I realized that using the native api was an easier approach. I suggest you to check the following two repos that helped me a lot while customizing my experiment:
https://github.com/CasellaJr/Benchmarking-FedAvg-and-FedCurv-for-Image-Classification-Tasks
https://github.com/CasellaJr/Benchmarking-Normalization-Layers-in-Federated-Learning-for-Image-Classification-Tasks-on-non-IID

Moreover, if you look through the discussions, you will find some questions that I have been encountering while creating my experiment and they might help you. Good luck!
